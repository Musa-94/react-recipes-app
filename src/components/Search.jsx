import React from 'react';

const Search = props => {
    const {
        search,
        handleSubmit,
        handleChange,
    } = props;

    return (
        <div className={'container'}>
            <div className={'row'}>
                <div className='col-10 mx-auto col-md-8 mt-5 text-center'>
                    <h1 className={'text-slanted text-capitalize'}>
                        search recipes with{' '}
                        <strong className={'text-orange'}> Food2Fork </strong>
                    </h1>
                    <form className='mt-4'>
                        <label htmlFor={'search'} className={'text-capitalize'}>
                            types recipes separated by comma
                        </label>
                        <div className={'input-group'}>
                            <input type={'text'}
                                   name={'search'}
                                   value={search}
                                   onChange={handleChange}
                                   className={'form-control'}
                                   placeholder={'chicken, onion, carrots'}
                            />
                            <div className='input-group-append'>
                                <button type={'submit'}
                                        onClick={handleSubmit}
                                        className={'input-group-text bg-primary text-white'}
                                >
                                    <i className={'fas fa-search'} />
                                </button>
                            </div>
                        </div>
                    </form>
                </div>
            </div>
        </div>
    );
}

export default Search;
