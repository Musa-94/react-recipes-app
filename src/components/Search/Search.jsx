import React from 'react';

const Search = props => {
    const {
        search,
        isError,
        handleSubmit,
        handleChange,
    } = props;

    return (
        <div className={'container'}>
            <div className={'row'}>
                <div className='col-10 mx-auto col-md-8 mt-5 text-center'>
                    <h1 className={'text-slanted text-capitalize'}>
                        search recipes with{' '}
                        <strong className={'text-orange'}> Food2Fork </strong>
                    </h1>
                    <form className='mt-4'>
                        <label htmlFor={'search'} className={'text-capitalize'}>
                            types recipes separated by comma
                        </label>
                        <div className={'input-group container col-sm'}>
                            <input type={'text'}
                                   name={'search'}
                                   value={search}
                                   onChange={event => handleChange(event)}
                                   className={'form-control'}
                                   placeholder={'chicken, onion, carrots'}
                            />
                            <div className='input-group-append'>
                                <button type={'submit'}
                                        onClick={handleSubmit}
                                        className={'input-group-text bg-primary text-white'}
                                >
                                    <i className={'fas fa-search'} />
                                </button>
                            </div>
                            {isError ?
                                <div className={'col-sm'}>
                                    <span children={'this recipe was not found'}
                                          className={'text-danger text-capitalize'}
                                    />
                                </div> : null
                            }
                        </div>
                    </form>
                </div>
            </div>
        </div>
    );
}

export default Search;
