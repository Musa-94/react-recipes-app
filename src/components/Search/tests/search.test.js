import React from 'react';
import Search from '../Search';

const props = {
    search: '',
    isError: false,
    handleSubmit: () => {},
    handleChange: () => {},
}

describe('Search snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<Search {...props}/>);

        expect(component).toMatchSnapshot();
    });
});