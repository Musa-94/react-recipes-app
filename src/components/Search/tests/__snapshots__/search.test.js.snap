// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot should correct renderer 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-10 mx-auto col-md-8 mt-5 text-center"
    >
      <h1
        className="text-slanted text-capitalize"
      >
        search recipes with
         
        <strong
          className="text-orange"
        >
           Food2Fork 
        </strong>
      </h1>
      <form
        className="mt-4"
      >
        <label
          className="text-capitalize"
          htmlFor="search"
        >
          types recipes separated by comma
        </label>
        <div
          className="input-group container col-sm"
        >
          <input
            className="form-control"
            name="search"
            onChange={[Function]}
            placeholder="chicken, onion, carrots"
            type="text"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="input-group-text bg-primary text-white"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="fas fa-search"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
